### PR TITLE
additional metrics for smartengine

### PR DESCRIPTION
**Pull request by davidbeesley**
- [x] Add max (in and out) to metric
- [x] Add unit tests

### CODE REVIEW

File: crates/fluvio-smartengine/src/engine.rs [hunk 1]
```diff
@@ -109,21 +110,26 @@ impl SmartModuleChainInstance {
...
         if let Some((last, instances)) = self.instances.split_last_mut() {
+            metric.add_records_in(number_of_records);
             let mut next_input = input;
 
             for instance in instances {
                 // pass raw inputs to transform instance
                 // each raw input may result in multiple records
                 let output = instance.process(next_input, &mut self.store)?;
+                smartmodule_usage_out += output.successes.len() as u64;
+                smartmodule_usage_in += output.successes.len() as u64;
```
Comment (sehz) [inline crates/fluvio-smartengine/src/engine.rs:132]: why increase this? this is output

Comment (davidbeesley) [reply crates/fluvio-smartengine/src/engine.rs:132]: because the output of previous is the input to next

Comment (davidbeesley) [reply crates/fluvio-smartengine/src/engine.rs:132]: notice this is line is not called for the last smartmodule.

Comment (sehz) [reply crates/fluvio-smartengine/src/engine.rs:132]: what's definition of usage then?

Comment (davidbeesley) [reply crates/fluvio-smartengine/src/engine.rs:132]:
```
 usage = max(total_records_in, total_records_out) where
 total_records_in = Number of records passed to each smartmodule in the chain.
 total_records_out =Number of records returned from each smartmodule in the chain.
```

Comment (sehz) [reply crates/fluvio-smartengine/src/engine.rs:132]: Suppose we have the following records count of each stage 10 => 5 => 3 => 2,  total amount of work should be just the sum of all inputs = 10+5+3 = 18

Comment (davidbeesley) [reply crates/fluvio-smartengine/src/engine.rs:132]: intent is to capture case where more output than input; 
For example 1 => 10.

Comment (sehz) [reply crates/fluvio-smartengine/src/engine.rs:132]: In that case 10 => 5 => 3 => 10,  so total record processed = 10 + 5 + 10.  So the correct definition is
total usage  = sum ( usage in each stage).   usage in each stage = max(input,output)

**Comment by davidbeesley** [comment]
No longer necessary.

**Pull request closed without merging**